sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -528,6 +536,56 @@ public static Schema from(StructType input) {
     return builder.build();
   }
 
+  /**
+   * Converts a list of spark data types to a cudf schema.
+   *
+   * <br/>
+   *
+   * This method correctly handles nested types, but will generate random field names.
+   *
+   * @param dataTypes the list of data types to convert
+   * @return the cudf schema
+   */
+  public static Schema from(DataType[] dataTypes) {
+    Schema.Builder builder = Schema.builder();
+    visit(dataTypes, builder, 0);
+    return builder.build();
+  }
+
+  private static void visit(DataType[] dataTypes, Schema.Builder builder, int level) {
+    for (int idx = 0; idx < dataTypes.length; idx ++) {
+      DataType dt = dataTypes[idx];
+      String name = "_col_" + level + "_" + idx;
```

Comment:
Personally I'd prefer 1, especially since cudf doesn't care about column names in most contexts.  cc: @revans2 and @abellina who may have a strong opinion on this.